executor/runtime/types/types.go

```diff
@@ -147,51 +125,97 @@ type Container struct {
+	LogStdioCheckInterval time.Duration
+	// LogLocalFileAfterUpload indicates whether or not we should delete log files after uploading them
+	LogKeepLocalFileAfterUpload bool
+	LogUploadRegexp             *regexp.Regexp
+	UploaderConfig              uploader.Config
 
 	pod *corev1.Pod
 
 	// GPU devices
 	gpuInfo GPUContainer
-	// Set if using a non-runc runtime to run system service init commands
-	runtime string
 
-	iamRole string
+	IamRole             string
+	imageName           string
+	imageFullyQualified string
+	imageDigest         string
+	imageVersion        string
+
+	JobGroupStack    string
+	JobGroupDetail   string
+	JobGroupSequence string
+	AppName          string
+
+	entrypoint []string
+	command    []string
+
+	AllowCPUBursting bool
+	batchEnabled     bool
+	batchPriority    string
+	capabilities     *titus.ContainerInfo_Capabilities
+	efsConfig        []*titus.ContainerInfo_EfsConfigInfo
+	hostnameStyle    string
+	// Is this container meant to run SystemD?
+	IsSystemD       bool
+	KillWaitSeconds uint32
+	// FuseEnabled determines whether the container has FUSE devices exposed to it
+	FuseEnabled bool
+	// KvmEnabled determines whether the container has KVM exposed to it
+	KvmEnabled         bool
+	MetatronCreds      *titus.ContainerInfo_MetatronCreds
+	OomScoreAdjEnabled bool
+	OomScoreAdj        int32
+	// Set if using a non-runc runtime to run system service init commands
+	runtime            string
+	RequireIMDSToken   string
+	ServiceMeshEnabled bool
+	ServiceMeshImage   string
+	// ShmSize sets the container's /dev/shm size
+	ShmSizeMiB uint32
+	TTYEnabled bool
 
 	Config config.Config
 }
 
-func (c *Container) VPCAccountID() string {
-	if vpcAccountID, ok := c.TitusInfo.GetPassthroughAttributes()[AccountIDParam]; ok {
-		return vpcAccountID
-	}
+func (c *Container) GetCapabilities() *titus.ContainerInfo_Capabilities {
+	return c.capabilities
+}
 
-	// If the param wasn't passed via pass through attributes, then fall back to pulling it from the host env
-	return c.Config.EC2AccountID
+func (c *Container) GetEfsConfigInfo() []*titus.ContainerInfo_EfsConfigInfo {
+	return c.efsConfig
 }
 
 // combineAppStackDetails is a port of the method with the same name from frigga.
 // See: https://github.com/Netflix/frigga/blob/v0.17.0/src/main/java/com/netflix/frigga/NameBuilder.java
-func combineAppStackDetails(taskInfo *titus.ContainerInfo) string {
-	var (
-		stack   = taskInfo.GetJobGroupStack()
-		details = taskInfo.GetJobGroupDetail()
-		appName = taskInfo.GetAppName()
-	)
-	if details != "" {
-		return fmt.Sprintf("%s-%s-%s", appName, stack, details)
+func combineAppStackDetails(c *Container) string {
+	if c.JobGroupDetail != "" {
+		return fmt.Sprintf("%s-%s-%s", c.AppName, c.JobGroupStack, c.JobGroupDetail)
 	}
-	if stack != "" {
-		return fmt.Sprintf("%s-%s", appName, stack)
+	if c.JobGroupStack != "" {
```

Comment:
Does it make sense to have `combinedAppStackDetails` as a receiver method of Container?